dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/registry/ZoneAwareClusterInvoker.java

```diff
@@ -148,60 +146,55 @@ public Result doInvoke(Invocation invocation, final List<Invoker<T>> invokers, L
-            if (null == rule) {
+
+            if (rule == null) {
                 rule = migrationClusterInvoker.getMigrationRule();
-            } else {
-                // 不一致
-                if (!rule.equals(migrationClusterInvoker.getMigrationRule())) {
-                    rule = MigrationRule.queryRule();
-                    break;
-                }
+                continue;
+            }
+
+            // 不一致
+            if (!rule.equals(migrationClusterInvoker.getMigrationRule())) {
+                rule = MigrationRule.queryRule();
+                break;
             }
         }
 
         MigrationStep step = rule.getStep();
 
         switch (step) {
             case FORCE_INTERFACE:
-                clusterRefresh(addreddChanged, interfaceInvokers);
-                clusterDestory(addreddChanged, serviceInvokers, true);
+                clusterRefresh(addressChanged, interfaceInvokers);
+                clusterDestroy(addressChanged, serviceInvokers, true);
                 if (logger.isDebugEnabled()) {
                     logger.debug("step is FORCE_INTERFACE");
                 }
                 return interfaceInvokers;
 
             case APPLICATION_FIRST:
-                clusterRefresh(addreddChanged, serviceInvokers);
-                clusterRefresh(addreddChanged, interfaceInvokers);
+                clusterRefresh(addressChanged, serviceInvokers);
+                clusterRefresh(addressChanged, interfaceInvokers);
 
-                if (serviceInvokers.size() > 0) {
-                    if (shouldMigrate(addreddChanged, serviceInvokers, interfaceInvokers)) {
-                        //clusterDestory(addreddChanged, interfaceInvokers, false);
+                boolean serviceAvailable = !serviceInvokers.isEmpty();
+                if (serviceAvailable) {
+                    if (shouldMigrate(addressChanged, serviceInvokers, interfaceInvokers)) {
                         if (logger.isDebugEnabled()) {
                             logger.debug("step is APPLICATION_FIRST shouldMigrate true get serviceInvokers");
                         }
                         return serviceInvokers;
-
-                    } else {
-                        if (logger.isDebugEnabled()) {
-                            logger.debug("step is APPLICATION_FIRST shouldMigrate false get interfaceInvokers");
-                        }
-                        return interfaceInvokers;
-                    }
-                } else {
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("step is APPLICATION_FIRST serviceInvokers is empty get interfaceInvokers");
                     }
-                    return interfaceInvokers;
                 }
 
+                if (logger.isDebugEnabled()) {
+                    logger.debug("step is APPLICATION_FIRST serviceInvokers " + ((!serviceAvailable)? "is empty": "shouldMigrate false") + " get interfaceInvokers");
```

Comment:
`serviceInvokers` should be judge by `serviceAvailable`.The logic isn't same as before.

Reply:
`serviceInvokers.size() > 0` equals `!serviceInvokers.isEmpty()` 
`serviceAvailable` equals `serviceInvokers.size() > 0`